### PR TITLE
feat(node/assert): add deepEqual and notDeepEqual 

### PR DESCRIPTION
**Pull request by tjosepo**
Those were skipped because they don't have test cases in `test-assert.js`.

I used the implementation of `isDeepEqual()` from [`./node/internal/utils/comparisons.ts`](https://github.com/denoland/deno_std/blob/main/node/internal/util/comparisons.ts), which I believe to be a direct copy of the code in Node.

These functions missing is what prevents [Recast](https://github.com/benjamn/recast) from working on Deno using esm.sh.

It's my first time contributing to the Node compat modules, so let me know if I missed anything 😊

### CODE REVIEW

File: node/assert.ts [hunk 1]
```diff
@@ -317,21 +318,63 @@ function notStrictEqual(
...
+    throw message;
+  }
+
+  throw createAssertionError({
+    actual,
+    expected,
+    message,
+    operator: "notDeepEqual",
+    generatedMessage: true,
+  });
```
Comment (cjihrig) [inline node/assert.ts:377]: Is it possible to avoid all of this duplication. Look at the `innerFail()` function in Node's `assert` library as an example.

**Review by cjihrig — COMMENTED** [review]
The changes LGTM, but it would be nice to cut down on some code duplication.

**Review by cjihrig — COMMENTED** [review]
The changes LGTM, but it would be nice to cut down on some code duplication.

**Comment by tjosepo** [comment]
Thank you for the review!

I added the `innerFail()` function.

**Review by cjihrig — APPROVED** [review]
LGTM. Thanks for the contribution.

**Pull request merged**